ddl/db_test.go

```diff
@@ -2340,6 +2340,14 @@ func (s *testDBSuite4) TestChangeColumn(c *C) {
 	c.Assert(hasNotNull, IsFalse)
 	// for enum
 	s.mustExec(tk, c, "alter table t3 add column en enum('a', 'b', 'c') not null default 'a'")
+	// https://github.com/pingcap/tidb/issues/23488
+	// if there is a prefix index on the varchar column, then we can change it to text
+	s.mustExec(tk, c, "drop table if exists t")
+	s.mustExec(tk, c, "create table t (k varchar(10), v int, INDEX(k(7)));")
```

Comment:
```suggestion
	s.mustExec(tk, c, "create table t (k char(10), v int, INDEX(k(7)));")
```
The conversion between varchar and non-varchar types has been forbidden in https://github.com/pingcap/tidb/pull/23688...

We can use `char` here instead for the test.

Reply:
Thanks, now all passed.